src/main/java/com/cloudbees/jenkins/support/slowrequest/SlowRequestThreadDumpsGenerator.java

```diff
@@ -110,17 +117,35 @@ public void run() {
                 fileNameDate += FREQUENCY_SEC * 1000;
             }
         }
+
+        setRunningStatus(false);
     }
 
     public static synchronized boolean checkThreadDumpsTrigger(long iota) {
         boolean newThreadDumps = false;
-        if (latestGeneratedSlowRequestThreadDump == 0l
-                || iota - latestGeneratedSlowRequestThreadDump > RECURRENCE_PERIOD_MILLIS) {
+
+        if (!isRunning()
+                && (latestGeneratedSlowRequestThreadDump == 0l
+                        || iota - latestGeneratedSlowRequestThreadDump > RECURRENCE_PERIOD_MILLIS)) {
             newThreadDumps = true;
             latestGeneratedSlowRequestThreadDump = iota;
         }
         return newThreadDumps;
     }
 
+    private static boolean isRunning() {
+        boolean runningLocal;
+        synchronized (SlowRequestThreadDumpsGenerator.class) {
+            runningLocal = running;
+        }
+        return runningLocal;
+    }
+
+    private static void setRunningStatus(boolean runningLocal) {
+        synchronized (SlowRequestThreadDumpsGenerator.class) {
+            running = runningLocal;
+        }
+    }
```

Comment:
Could be simplified using `AtomicBoolean`.

Reply:
Hi @Dohbedoh,

Thank you for the tip. I have never used AtomicBoolean, so I read the documentation about the topic, and I performed some changes to the PR. Could you please revisit it and confirm if the usage of AtomicBoolean is correct?

Kind regards,